client/state/terms/actions.js

```diff
@@ -1,15 +1,77 @@
 /**
+ * External dependencies
+ */
+import omit from 'lodash/omit';
+
+/**
  * Internal dependencies
  */
 import wpcom from 'lib/wp';
 import {
+	TERMS_ADD_REQUEST,
+	TERMS_ADD_REQUEST_SUCCESS,
+	TERMS_ADD_REQUEST_FAILURE,
 	TERMS_RECEIVE,
 	TERMS_REQUEST,
 	TERMS_REQUEST_SUCCESS,
 	TERMS_REQUEST_FAILURE
 } from 'state/action-types';
 
 /**
+ * Module constants
+ */
+let temporaryIdCount = 0;
+
+function getTemporaryId() {
+	const temporaryId = [ 'temporary', temporaryIdCount ].join( '-' );
+	temporaryIdCount++;
+	return temporaryId;
+}
+
+/**
+ * Returns an action thunk, dispatching progress of a request to add a new term
+ * the site and taxonomy.
+ *
+ * @param  {Number} siteId   Site ID
+ * @param  {String} taxonomy Taxonomy Slug
+ * @param  {Object} args     Object of new term attributes
+ * @return {Object}          Action object
+ */
+export function addTerm( siteId, taxonomy, args ) {
+	return ( dispatch ) => {
+		const temporaryId = getTemporaryId();
+
+		dispatch( {
+			type: TERMS_ADD_REQUEST,
+			temporaryId,
+			siteId,
+			taxonomy,
+			args
+		} );
+
+		return wpcom.site( siteId ).taxonomy( taxonomy ).term().add( args ).then( ( data ) => {
```

Comment:
If you're feeling adventurous, maybe explore some `async` / `await` magic:

```js
return async ( dispatch ) => {
	// ...

	try {
		const data = await wpcom.site( siteId ).taxonomy( taxonomy ).term().add( args );
		dispatch( {
			type: TERMS_ADD_REQUEST_SUCCESS,
			data: omit( data, '_headers' ),
			temporaryId,
			siteId,
			taxonomy,
			args
		} );
	} catch ( error ) {
		dispatch( {
			type: TERMS_ADD_REQUEST_FAILURE,
			temporaryId,
			siteId,
			taxonomy,
			args,
			error
		} );
	}
}
```

🔮